built_in_tasks/target_capture_task.py

```diff
@@ -655,7 +672,41 @@ def rand_target_chain_2D(ntrials=100, chain_length=1, boundaries=(-12,12,-12,12)
             pts = pts+(boundaries[0], 0, boundaries[2])
             yield idx+np.arange(chain_length), pts
             idx += chain_length
-    
+
+    @staticmethod
+    def rand_same_target_chain_2D(nblocks=10, num_same_trials=10, chain_length=1, boundaries=(-12,12,-12,12)):
+        '''
+        Generates a sequence of 2D (x and z) target pairs. The same target appears num_same_trials times.
+
+        Parameters
+        ----------
+        nblocks : int
+            The number of different target positions
+        num_same_trials : int
+            The number of trials for the same target position shown in a row
+        chain_length : int
+            The number of targets in each chain
+        boundaries: 4 element Tuple
+            The limits of the allowed target locations (-x, x, -z, z)
+
+        Returns
+        -------
+        [ntrials x chain_length x 3] array of target coordinates
+        '''
+
+        rng = np.random.default_rng()
+        idx = 0
+        for iblock in range(nblocks):
+            # Choose a random sequence of points within the boundaries
+            pts = rng.uniform(size=(chain_length, 3))*((boundaries[1]-boundaries[0]),
+                0, (boundaries[3]-boundaries[2]))
+            pts = pts+(boundaries[0], 0, boundaries[2])
+
+            # Generate the same target trial_streak times
+            for itrial in range(num_same_trials):
+                yield idx+np.arange(chain_length), pts
+                idx += chain_length
```

Comment:
i think you only want to increment idx every time you generate new pts, not for every repetition of the same target